packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -137,7 +137,10 @@ enum HeroFlightDirection {
 /// is, by default, used to do the transition: when going from route A to route
 /// B, route B's hero's widget is placed over route A's hero's widget. If a
 /// [flightShuttleBuilder] is supplied, its output widget is shown during the
-/// flight transition instead.
+/// flight transition instead. If overriding [flightShuttleBuilder] leads to
+/// gaps or jumps arising in the animation, try interpolating between
```

Comment:
There could be other [InheritedWidget](https://master-api.flutter.dev/flutter/widgets/InheritedWidget-class.html)s (including custom/3rd party ones) other than `MediaQuery` that may cause contents in the `Hero` to have unexpected padding/changes appearance during the Hero animation. Instead of directly suggesting `MediaQuery` is the culprit I think adding a hint so that  developer are aware of the general problem with `InheritedWidget`s in a `Hero`, and maybe use MediaQuery as an example would be a bit more helpful.

maybe something along the lines of (rephrase as you see fit):

The flightShuttleBulder tree uses [InheritedWidget]s from the route the Hero animation starts from (I don't remember if this statement is true, could be the to route?)
If the widget tree in flightShuttleBuilder changes its geometry/appearance based on [InheritedWidget]s that's above the Hero widget tree, for example if both routes provide `MediaQuery` but with different paddings, you may see jumps/gaps. If you experience the problem, wrap the current flightShuttleBulder in a such inherited widget (and potentially interpolate the value provided by the inherited widget using the `animation` parameter. The default flightShuttleBulder is wrapped in a MediaQuery widget if only one navigator has padding or both navigator's `MediaQuery`s provide different paddings. 

Also you may want to add docs to [flightShuttleBulder] itself too.

Reply:
@LongCatIsLooong 's change make it much clearer. Although as I said, I think this is better moved to the doc of `flightShuttleBulder`. 

For the last sentence: "The default flightShuttleBulder is wrapped in a MediaQuery widget if only both navigators' MediaQuerys provide different paddings. " should be enough.

Reply:
Ok, if I understand https://github.com/flutter/flutter/pull/105654/files#r893811742 correctly, the jump will occur if the source tree and the target tree uses different custom `InheritedWidget` settings even without `flightShuttleBulder`. So I suggest (instead of moving) changing the doc starting " If a [flightShuttleBuilder] is supplied...“ into the follows:

However, if the hero uses information from `InheritedWidget`s, and the inherited data are different between the starting tree and the target tree, there will be an ugly jumping animation unless that data is properly interpolated. By default, `Hero` interpolates `MediaQuery`'s paddings. If your hero uses custom `InheritedWidget`s that see jumping animation, try to provide custom in-flight transition using `flightShuttleBuilder`.